### ISSUE TITLE
[RFC] Remove `X-APOLLO-...` custom headers

### ISSUE THREAD

**Issue by martinbonnin** [apollographql/apollo-kotlin#5345]
### Description

Apollo Kotlin currently sends 2 extra headers by default with each request:

* `X-APOLLO-OPERATION-NAME`
* `X-APOLLO-OPERATION-ID`

In order to simplify the code as well as align with other clients, we'd like to remove them. 

The only usage we are aware of is to collaborate with [CSRF prevention in Apollo Server](https://www.apollographql.com/docs/apollo-server/security/cors/#preventing-cross-site-request-forgery-csrf) but:
1. the conditions where it's actually useful are pretty narrow: you'd need to use Apollo Server and GraphQL File Uploads (which [we discourage](https://www.apollographql.com/blog/backend/file-uploads/file-upload-best-practices/))
2. the mitigation is as easy as adding a `Apollo-Require-Preflight` header

More generally, it's easy to add any header back if required:

```kotlin
  val apolloClient = ApolloClient.Builder()
      .addInterceptor(object : ApolloInterceptor {
        override fun <D : Operation.Data> intercept(request: ApolloRequest<D>, chain: ApolloInterceptorChain): Flow<ApolloResponse<D>> {
          return chain.proceed(request.newBuilder().addHttpHeader("X-APOLLO-REQUEST-NAME", request.operation.name()).build())
        }
      })
      .build()
```

Overall, we feel this is a win for the vast majority of users and the mitigation is quite easy for the others. If you feel otherwise, please comment. Feedbacks welcome! 
Timeline is doing it for v4 by the end of the year.

**Comment by glasser** [comment]
This will mean that the default behavior of Apollo Router and Apollo Server will be to block all GET requests from Apollo Kotlin. I'd encourage you to pair this with adding `apollo-require-preflight` to all GET requests.

**Comment by glasser** [comment]
(Previous comment was a response to the original version of the issue which has been edited. Note that this applies to Apollo Router as well as Apollo Server.)

**Comment by martinbonnin** [comment]
> (Previous comment was a response to the original version of the issue which has been edited. Note that this applies to Apollo Router as well as Apollo Server.)

@glasser Yup, apologies about the edits and thank you so much for surfacing this.  Initial post edited to:

* mention `GET` requests
* mention `Apollo-Require-Preflight` (any preference about the casing here?) for both `GET` and uploads
* mention Apollo Router in addition to Apollo Server.
